tests/Feature/RegisterTest.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+namespace Tests\Feature;
+
+use Tests\TestCase;
+use App\Models\User;
+use Firebase\JWT\JWT;
+use App\Traits\InteractsWithJWT;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+
+class RegisterTest extends TestCase
+{
+    use RefreshDatabase, InteractsWithJWT;
+
+    /**
+     * @var string
+     */
+    protected $route;
+
+    /**
+     * Setup the test environment.
+     *
+     * @return void
+     */
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->route = route('auth.register');
+    }
+
+    /**
+     * @param  array  $override
+     * @return void
+     */
+    private function assertValidationError($override = [])
+    {
+        $response = $this->postJson($this->route, $this->validParamaters($override));
+
+        $response->assertStatus(422);
+
+        if (array_key_exists('password_confirmation', $override)) {
+            unset($override['password_confirmation']);
+        }
```

Comment:
These 3 lines could instead be a single line: `Arr::forget($override, 'password_confirmation')`. If the list were to ever grow that second parameter could be given an array instead.

Reply:
Ah nice, I like this better.